### PR TITLE
tests: create test user with useradd in uc24

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change is needed to create the test user in uc24 with home directory.

This is needed to fix tests in uc24 (edge validation)

### CODE REVIEW

File: tests/lib/prepare-restore.sh [hunk 1]
```diff
@@ -293,7 +293,11 @@ prepare_project() {
     fi
 
     if [ "$SPREAD_BACKEND" = "testflinger" ]; then
-        adduser --uid 12345 --extrausers --quiet --disabled-password --gecos '' test
+        if os.query is-core-ge 24; then
+            useradd --uid 12345 -m --extrausers test
+        else
+            adduser --uid 12345 --extrausers --quiet --disabled-password --gecos '' test
+        fi
```
Comment (olivercalder) [inline tests/lib/prepare-restore.sh:300]: Is the `--disabled-password` essential, and is this the default behavior for `useradd`? I don't see any equivalent option there.

Also, the `--gecos` option is deprecated, and `--comment` should be used instead. But if the comment is empty, is `--gecos` even necessary?

>        --comment comment
              Set the comment field for the new entry generated.  adduser will not ask for the information  if  this  option  is
              given.  This field is also known under the name GECOS field and contains information that is used by the finger(1)
              command.   This  used  to  be  the  --gecos option, which is deprecated and will be removed after Debian bookworm.

Comment (olivercalder) [inline tests/lib/prepare-restore.sh:300]: Is the `--disabled-password` essential, and is this the default behavior for `useradd`? I don't see any equivalent option there.

Also, the `--gecos` option is deprecated, and `--comment` should be used instead. But if the comment is empty, is `--gecos` even necessary?

```
       --comment comment
              Set the comment field for the new entry generated.  adduser will not ask for the information  if  this  option  is
              given.  This field is also known under the name GECOS field and contains information that is used by the finger(1)
              command.   This  used  to  be  the  --gecos option, which is deprecated and will be removed after Debian bookworm.
```

File: tests/lib/nested.sh [hunk 2]
```diff
@@ -167,8 +167,8 @@ nested_uc20_transition_to_system_mode() {
 
 nested_prepare_ssh() {
     if nested_is_core_ge 24; then
-        remote.exec "sudo useradd --uid 12345 --extrausers test"
-        remote.exec "sudo useradd --extrausers external"
+        remote.exec "sudo useradd --uid 12345 -m --extrausers test"
+        remote.exec "sudo useradd -m --extrausers external"
     else 
         remote.exec "sudo adduser --uid 12345 --extrausers --quiet --disabled-password --gecos '' test"
         remote.exec "sudo adduser --extrausers --quiet --disabled-password --gecos '' external"
```
Comment (olivercalder) [inline tests/lib/nested.sh:174]: The `--gecos` option is deprecated, replaced by `--comment`. But if comment is empty, could the option be removed entirely?

**Review by olivercalder — APPROVED** [review]
Thanks! I was wary of using `--extrausers` with `useradd` instead of `adduser`, since it's not mentioned in the manpage for `useradd`, but `useradd --help` shows it as an option. So this looks good to me.

Note that `--gecos` option is deprecated in `adduser`, perhaps that could be replaced now as well?

**Review by andrewphelpsj — APPROVED** [review]
thanks!

File: tests/lib/nested.sh [hunk 3]
```diff
@@ -167,8 +167,8 @@ nested_uc20_transition_to_system_mode() {
 
 nested_prepare_ssh() {
     if nested_is_core_ge 24; then
-        remote.exec "sudo useradd --uid 12345 --extrausers test"
-        remote.exec "sudo useradd --extrausers external"
+        remote.exec "sudo useradd --uid 12345 -m --extrausers test"
```
Comment (zyga) [inline tests/lib/nested.sh:170]: Nitpick, I'd expand -m to --create-home so that it's more obvious what's going on.

**Review by zyga — APPROVED** [review]
Looks good, one inline remark about -m

**Pull request merged**